src/css3_parser/parser.rs

```diff
@@ -0,0 +1,500 @@
+use super::{
+    tokenizer::CSSTokenizer,
+    tokens::{Token, TokenType},
+};
+
+#[derive(Debug, PartialEq)]
+pub struct CSSStyleSheet {
+    css_rules: Vec<CSSRule>,
+}
+
+impl Default for CSSStyleSheet {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl CSSStyleSheet {
+    pub fn new() -> CSSStyleSheet {
+        CSSStyleSheet {
+            css_rules: Vec::new(),
+        }
+    }
+}
+
+pub type CSSPropName = String;
+pub type CSSPropValue = String;
+
+#[derive(Debug, PartialEq)]
+pub struct CSSStyleDeclaration {
+    name: CSSPropName,
+    value: CSSPropValue,
+}
+
+#[derive(Debug, PartialEq)]
+pub struct CSSSelector {
+    path: Vec<String>,
+}
+
+impl Default for CSSSelector {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl CSSSelector {
+    pub fn new() -> CSSSelector {
+        CSSSelector { path: Vec::new() }
+    }
+
+    pub fn from_str_slice(path: Vec<&str>) -> CSSSelector {
+        CSSSelector {
+            path: path.iter().map(|s| s.to_string()).collect::<Vec<String>>(),
+        }
+    }
+
+    pub fn add(&mut self, value: String) -> &mut CSSSelector {
+        self.path.push(value);
+
+        self
+    }
+}
+
+#[derive(Debug, PartialEq)]
+pub struct CSSStyleRule {
+    selector: CSSSelector,
+    style_declarations: Vec<CSSStyleDeclaration>,
+}
+
+#[derive(Debug, PartialEq)]
+pub struct CSSMeidaRule {
```

Comment:
I just noticed this now, is this supposed to me `Meida` or `Media`?

Reply:
Do we need to fix this before merge? Doesn't seem a blocker

Reply:
This is easy to fix later on.  But eventually we'll want to block on this kind of thing, once the rate of PRs slows down or there's more reviewers.

Reply:
Agree, and since the CSS parser won’t even be used for a bit/in hot development, I don’t think it’s a big deal to fix right away, maybe just create an issue so we don’t forget